File: app/src/main/java/com/code_design_camp/client/friday/HeadDisplayClient/fragments/store/MainStoreFragment.java
package com.code_design_camp.client.friday.HeadDisplayClient.fragments.store;


import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.IdRes;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;

import com.code_design_camp.client.friday.HeadDisplayClient.R;
import com.code_design_camp.client.friday.HeadDisplayClient.fragments.net.ConnectionFragment;
import com.code_design_camp.client.friday.HeadDisplayClient.fragments.net.OnConnectionStateChangedListener;
import com.code_design_camp.client.friday.HeadDisplayClient.store.fragments.StoreFeaturedFragment;

public class MainStoreFragment extends ConnectionFragment implements OnConnectionStateChangedListener {


    public MainStoreFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View v = inflater.inflate(R.layout.fragment_main_store, container, false);
        Toolbar toolbar = v.findViewById(R.id.store_toolbar);
        toolbar.setTitle("");
        ((AppCompatActivity) getActivity()).setSupportActionBar(toolbar);
        getActivity().invalidateOptionsMenu();
        addFragment(R.id.store_featured_fragment_container, new StoreFeaturedFragment());
        return v;
    }

    @Override
    public void onConnected() {

    }

    @Override
    public void onError(Exception e) {
        Log.e("StoreError", "Error occured at store");
    }

    private void addFragment(@IdRes int containerId, Fragment newFragment) {
        getChildFragmentManager().beginTransaction()
                .replace(containerId, newFragment)
                .commit();
    }
}

File: appModules/store/src/main/java/com/friday/ar/store/ui/fragments/MainStoreFragment.kt
package com.friday.ar.store.ui.fragments


import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.lifecycle.Observer
import com.friday.ar.core.net.ConnectionFragment
import com.friday.ar.core.net.OnConnectionStateChangedListener
import com.friday.ar.store.R
import com.friday.ar.store.ui.fragments.feed.gridFragments.SimpleGridFragment
import extensioneer.notNull
import org.koin.android.viewmodel.ext.android.viewModel


class MainStoreFragment : ConnectionFragment(), OnConnectionStateChangedListener {
    companion object {
        private const val LOGTAG = "MainStoreFragment"
    }

    val storeViewModel by viewModel<StoreViewModel>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        Log.d("storefragment", "creating store view")
        // Inflate the layout for this fragment
        val v = inflater.inflate(R.layout.fragment_main_store, container, false)
        val toolbar = v.findViewById<Toolbar>(R.id.store_toolbar)
        toolbar.title = ""
        (activity as AppCompatActivity).setSupportActionBar(toolbar)
        activity!!.invalidateOptionsMenu()

        storeViewModel.structureData().observe(this, Observer { itemList ->
            itemList.forEach { document ->
                Log.d(LOGTAG, "found feed item: [id:${document.id}; type=${document.getString("type")}; title=${document.getString("title")}]")
                val simpleGridFragment = SimpleGridFragment.from(document)
                simpleGridFragment.notNull {
                    this@MainStoreFragment.childFragmentManager.beginTransaction()
                            .add(R.id.contentStore, this)
                            .addToBackStack(null)
                            .commit()
                }
            }
        })
        return v
    }

    override fun onConnected() {
        //if (childFragmentManager.backStackEntryCount != 0) childFragmentManager.popBackStack()
    }

    override fun onError(e: Exception) {
        Log.e("StoreError", "Error occured at store: ${e.message}")
    }
}
